coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt

```diff
@@ -135,67 +123,95 @@ fun CoilImage(
     loading: @Composable (() -> Unit)? = null,
     onRequestCompleted: (RequestResult) -> Unit = emptySuccessLambda
 ) {
-    WithConstraints(modifier) {
-        val requestWidth = constraints.requestWidth.value
-        val requestHeight = constraints.requestHeight.value
+    var result by state<RequestResult?> { null }
 
-        // Execute the request using executeAsComposable(), which guards the actual execution
-        // so that the request is only run if the request changes.
-        val result = when {
-            request.sizeResolver != null -> {
-                // If the request has a sizeResolver set, we just execute the request as-is
-                request.executeAsComposable()
-            }
-            requestWidth > 0 && requestHeight > 0 -> {
-                // If we have a non-zero size, we can modify the request to include the size
-                request.newBuilder()
-                    .size(requestWidth, requestHeight)
-                    .build()
-                    .executeAsComposable()
+    val requestActor = remember(request) {
+        CoilRequestActor(request) { result = it }
+    }
+
+    launchInComposition(requestActor) {
+        // Launch the Actor
+        requestActor.run()
+    }
+
+    onDispose {
+        // When we leave the composition, close the Actor channel
+        requestActor.close()
```

Comment:
you shouldn't need this since the requestActor uses consumeAsFlow - cancelling the call to run() will cancel the channel. Even that is probably unnecessary.

Reply:
Agreed, I also don't think it's safe to `consumeAsFlow()` because the Actor lifetime could (maybe) be longer than the composition. Moved to use `receiveAsFlow()` instead.